internal/logging/defaults.go

```diff
@@ -107,25 +107,27 @@ func (l *fileHandler) Emit(ctx *MessageContext, message string, args ...interfac
 	filename := filepath.Join(datadir, FileName())
 
 	// only log to rollbar when on release, beta or unstable branch and when built via CI (ie., non-local build)
-	if ctx.Level == "ERROR" && (constants.BranchName == constants.ReleaseBranch || constants.BranchName == constants.BetaBranch || constants.BranchName == constants.ExperimentalBranch) && rtutils.BuiltViaCI {
-		data := map[string]interface{}{}
-
-		if l.file != nil {
-			if err := l.file.Close(); err != nil {
-				data["log_file_close_error"] = err.Error()
-			} else {
-				logData, err := ioutil.ReadFile(filename)
-				if err != nil {
-					data["log_file_read_error"] = err.Error()
+	defer func() { // defer so that we can ensure errors are logged to the rollbar even if rollbar panics (which HAS happened!)
```

Comment:
I don't understand how this would ensure that the rollbar message doesn't panic.

IIUC the problem is that the problem actually is that the `m.Ready()` function is not interrupted right now and therefore can just hang out in the background. And therefore that background go process can call a rollbar error after we call `rollbar.Close`.

Reply:
It doesn't, this just ensures that we still have the error in the log file. Without this all we see in the log file is "rollbar panicked.. "

Reply:
```suggestion
	defer func() { // defer so that we can ensure errors are logged to the logfile even if rollbar panics (which HAS happened!)
```